### PR TITLE
Deenasun/week 2 assignment

### PR DESCRIPTION
**Pull request by deenasun**
[//]: # "Feel free to customize this template to your project's vibes"

### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
Modified `app/page.tsx`
- Added useState variables to store posts and comments
- Added useEffect to query posts and query comments from backend
- Mapped post and comment arrays to `Post` and `Comment` components

Modified styling for post, comment, and icons in `app/styles.module.css`

Added `components/Comment.tsx`
- Converts date string fetched from supabase into more readable date string format
- Displays comment with username, date, and comment

Added `components/InteractionBar.tsx`
- Displays the heart icon, number of likes, and share icon for the post

Added `components/Post.tsx`
- Created separate component for rendering the post
- Includes username, date, content, and image

Added `supabase/client.ts`
- Set up supabse client
- Wrote async functions to query posts and comments from supabase backend

Added `types/types.ts`
- Defined interfaces for component props and queried data

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

image

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

**Testing**
```
git fetch origin deenasun/branch1
git checkout deenasun/branch1
```

Run `pnpm run dev` in the terminal.

Navigate to `http://localhost:3000`

You should see the posts and comments show up on your screen!

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'
- https://supabase.com/docs/reference/javascript/introduction

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @ronniebeggs

### CODE REVIEW

File: supabase/client.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  const { data: posts, error } = await supabase.from('posts').select('*');
+
+  return { posts, error };
+}
+
+export async function queryComments() {
+  const { data: comments, error } = await supabase.from('comments').select('*');
+
+  return { comments, error };
+}
```
Comment (ronniebeggs) [inline supabase/client.ts:17]: It's sometimes helpful to initialize the supabase client in a separate file from the queries, but it doesn't matter for a project of this size.

File: components/Comment.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      <div className={styles.comment}>
+        <div className={styles.profile}>
+          <p className={styles.text}>{name}</p>
+          <p className={styles.date}>{commentDate}</p>
+        </div>
+        <p className={styles.commentText}>{comment}</p>
+      </div>
+    </div>
+  );
+}
```
Comment (ronniebeggs) [inline components/Comment.tsx:30]: Good job creating a separate component for comments! Your file and component organization is very clear.

File: app/page.tsx [hunk 3]
```diff
@@ -1,65 +1,51 @@
...
-
-        <div style={{ display: 'flex', justifyContent: 'space-between' }}>
-          <div style={{ display: 'flex', columnGap: '12px' }}>
-            <HeartIcon size={24} />
-            <p style={{ fontWeight: 600 }}>256 Likes</p>
-          </div>
-          <ShareIcon size={24} />
-        </div>
-      </div>
+      ))}
```
Comment (ronniebeggs) [inline app/page.tsx:39]: We only want to render one post per screen in future iterations of this project. However, given that there's only one post in the database right now, it doesn't really matter.

**Review by ronniebeggs — APPROVED** [review]
Great job Deena!!
